source/server/options_impl.cc

```diff
@@ -154,10 +155,9 @@ OptionsImpl::OptionsImpl(std::vector<std::string> args,
   TCLAP::MultiArg<std::string> stats_tag(
       "", "stats-tag",
       "This flag provides a universal tag for all stats generated by Envoy. The format is "
-      "``tag:value``. Only, alphanumeric values are allowed for stats and tags. This flag can be "
-      "repeated multiple times "
-      "to set multiple universal tags."
-      "However, multiple stats with different values are not allowed.",
+      "``tag:value``. Only alphanumeric values are allowed for tag names. For tag values all "
+      "characters are permitted except for '.' (dot). This flag can be repeated multiple times to "
+      "set multiple universal tags. However, multiple stats with different values are not allowed.",
```

Comment:
Maybe this is enough to establish the "envoy standard" but let's make sure @ggreenway and @snowp buy into this syntax.

I wasn't sure what you meant by the last sentence. Did you mean "multiple values for the same tag name are not allowed"?